docs/dev/development-process/development-process.rst

```diff
@@ -0,0 +1,220 @@
+This step requires careful adherence to the previously defined requirements and system architecture to ensure that the function fits seamlessly into the existing system and fulfills the specified functional and performance criteria.
+
+.. note::
+      Make sure to follow the `Artemis Code and Design Guidelines <https://docs.artemis.cit.tum.de/dev/guidelines.html>`_.
+
+5. Create a Pull Request
+========================
+After the feature implementation is complete, the developer is required to create a pull request for integrating the feature into the develop branch. 
+The subsequent sections provide guidance on the naming conventions and outline the necessary steps for creating and merging a pull request.
+
+Naming Conventions for GitHub Pull Requests
+-------------------------------------------
+
+1. The first term is a main feature of Artemis and should use code highlighting, e.g.  ``Programming exercises``:
+
+    1. Possible feature tags are: ``Programming exercises``, ``Integrated code lifecycle``, ``Quiz exercises``, ``Modeling exercises``, ``Text exercises``, ``File upload exercises``, ``Exam mode``,
+       ``Grading``, ``Assessment``, ``Communication``, ``Notifications``, ``Team exercises``, ``Lectures``, ``Integrated Markdown Editor``, ``Plagiarism checks``, ``Learning analytics``,
+       ``Adaptive learning``, ``Learning path``, ``Tutorial groups``, ``Iris``, ``Scalability``, ``Usability``, ``Performance``, ``Intrastructure``, ``Mobile Apps``.
+    2. If the change is not visible to end users, or it is a pure development or test improvement, we use the term ``Development``:.
+    3. Everything else belongs to the ``General`` category.
+
+2. The colon is not highlighted.
+
+3. After the colon, there should be a verbal form that is understandable by end users and non-technical persons, because this will automatically become part of the release notes.
+
+    1. The text should be short, non-capitalized (except the first word) and should include the most important keywords. Do not repeat the feature if it is possible.
+    2. We generally distinguish between bugfixes (the verb “Fix”) and improvements (all kinds of verbs) in the release notes. This should be immediately clear from the title.
+    3. Good examples:
+
+        - “Allow instructors to delete submissions in the participation detail view”
+        - “Fix an issue when clicking on the start exercise button”
+        - “Add the possibility for instructors to define submission policies”
+
+
+
+Steps to Create and Merge a Pull Request
+----------------------------------------
+
+0. Preconditions (For Developers Only)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+* Limit yourself to one functionality per pull request.
+* Split up your task in multiple branches & pull requests if necessary.
+* `Commit Early, Commit Often, Perfect Later, Publish Once. <https://speakerdeck.com/lemiorhan/10-git-anti-patterns-you-should-be-aware-of>`_
+
+1. Begin Implementation (For Developers Only)
```

Comment:
(Not for this PR but for future reference)
We should also discuss in the meeting if opening a draft PR at the start of development is in-line with how we handle PRs in the future. (See also [meeting notes](https://confluence.ase.in.tum.de/display/ArTEMiS/Artemis+Developer+Meeting+-+2024-04-22) from the last dev meeting)
![image](https://github.com/ls1intum/Artemis/assets/118574504/d640e0bd-ce83-404a-85db-6797ddb93095)